### PR TITLE
Export related objects

### PR DESCRIPTION
**Pull request by snazy**
Export and import objects related to the repository, the references, the commits and contents.

The IDs of the related objects to be exported are retrieved via `TransferRelatedObjects` implementations, introduced by #9033. Those are exported and imported using the ability to (de)serialize "any" custom object, introduced by #9032.

### CODE REVIEW

File: versioned/transfer/build.gradle.kts [hunk 1]
```diff
@@ -23,13 +23,19 @@ plugins {
 
 publishingHelper { mavenName = "Nessie - Import/Export" }
 
+sourceSets {
+  // This implicitly als adds all required Gradle tasks and configurations.
```
Comment (dimas-b) [inline versioned/transfer/build.gradle.kts:27]:
```suggestion
  // This implicitly also adds all required Gradle tasks and configurations.
```

File: versioned/transfer/src/main/java/org/projectnessie/versioned/transfer/ImportPersistV23.java [hunk 2]
```diff
@@ -129,4 +151,26 @@ void processCommit(Commit commit) throws ObjTooLargeException {
...
+
+    Obj obj;
+    if (type.equals(StandardObjType.UNIQUE)) {
+      obj = UniqueIdObj.uniqueId(id, genericObj.getSpace(), genericObj.getData());
+    } else {
+      ByteBuffer data = genericObj.getData().asReadOnlyByteBuffer();
+      String versionToken = genericObj.hasVersionToken() ? genericObj.getVersionToken() : null;
+
+      obj =
+          SmileSerialization.deserializeObj(
```
Comment (dimas-b) [inline versioned/transfer/src/main/java/org/projectnessie/versioned/transfer/ImportPersistV23.java:168]: Do we want to check the `GenericObj.encoding` value?

File: versioned/transfer/src/main/java/org/projectnessie/versioned/transfer/ProgressEvent.java [hunk 3]
```diff
@@ -28,6 +28,9 @@ public enum ProgressEvent {
   START_COMMITS,
   COMMIT_WRITTEN,
   END_COMMITS,
+  START_GENERIC,
```
Comment (dimas-b) [inline versioned/transfer/src/main/java/org/projectnessie/versioned/transfer/ProgressEvent.java:31]: This value does not appear to be used to produce exports events... should we note that in a comment, perhaps?

**Review by dimas-b — APPROVED** [review]
LGTM overall :+1:

File: versioned/transfer/src/main/java/org/projectnessie/versioned/transfer/ProgressEvent.java (hunk 3 above)
Comment (snazy) [reply versioned/transfer/src/main/java/org/projectnessie/versioned/transfer/ProgressEvent.java:31]: There's a comment in `org.projectnessie.tools.admin.cli.ExportRepository.ExportProgressListener#progress`

File: versioned/transfer-proto/src/main/proto/org/projectnessie/versioned/transfer/serialize/transfer.proto [hunk 4]
```diff
@@ -113,3 +125,24 @@ message RepositoryDescriptionProto {
...
+  // the short name or full name of the (custom) obj type
+  string type_name = 2;
+  bytes data = 3;
+  // UpdateableObj
+  optional string version_token = 4;
+  // custom obj
+  optional Encoding encoding = 5;
+  // the value of `o.p.versioned.storage.common.objtypes.Compression.valueString()`
+  optional string compression = 6;
+  // UniqueIdObj
```
Comment (adutra) [inline versioned/transfer-proto/src/main/proto/org/projectnessie/versioned/transfer/serialize/transfer.proto:146]: `UniqueIdObj` being a standard object type, isn't it better to create a separate proto for it?

File: versioned/transfer/src/main/java/org/projectnessie/versioned/transfer/ImportPersistV23.java (hunk 2 above)
Comment (snazy) [reply versioned/transfer/src/main/java/org/projectnessie/versioned/transfer/ImportPersistV23.java:173]: added

File: versioned/transfer-proto/src/main/proto/org/projectnessie/versioned/transfer/serialize/transfer.proto (hunk 4 above)
Comment (snazy) [reply versioned/transfer-proto/src/main/proto/org/projectnessie/versioned/transfer/serialize/transfer.proto:146]: Renamed the proto type :D

Comment (adutra) [reply versioned/transfer-proto/src/main/proto/org/projectnessie/versioned/transfer/serialize/transfer.proto:146]: Faire enough :-)

**Review by dimas-b — APPROVED** [review]

**Pull request merged**